### PR TITLE
Config system should flow insignificant properties

### PR DESCRIPTION
**Pull request by ericstj**
Some properties are "insignificant" which means that they aren't
considered for compatibility decisions but should still flow to child
projects.

In a normal build traversal these will flow because
FindBestConfiguration finds all compatible configurations for the
BuildConfiguration then chooses the best of these (which include
insignificant property values) from the set of project
BuildConfigurations.  As a result FindBestConfiguration will flow the
insignificant properties from BuildConfiguration to the Configuration
that gets passed to child projects.

This is different than BuildAllConfigurations.  In that case we don't
call any task to select configurations, we just repeat the
configurations verbatim from the project's BuildConfigurations.
Since projects never specify these insignificant properties
(eg: ConfigurationGroup, ArchGroup) we weren't flowing the values and
instead were always choosing the default value.

This change makes it so that we first try to pull insignificant
properties from the Configuration string, but if we can't find them,
we'll then try to persist the value (along with creating any derived
properties) and only if they are unset will we try to set them to their
default value.

This fixes an issue where the current BuildAllConfigurations legs
were always building debug regardless of -release being passed.

/cc @joperezr @dagood @chcosta @weshaggard

### CODE REVIEW

File: src/Tools/CoreFx.Tools/GenerateConfigurationProps.cs [hunk 1]
```diff
@@ -101,6 +101,20 @@ private void ParseProperties(string propertyName, ProjectRootElement project, bo
                     AddProperties(whenPropertiesElement, value, includeAdditionalProperites, parsedValuePrefix);
                 }
 
+                if (property.Insignificant)
```
Comment (joperezr) [inline src/Tools/CoreFx.Tools/GenerateConfigurationProps.cs]: Where are you flagging which properties are "insignificant"?

Comment (ericstj) [inline src/Tools/CoreFx.Tools/GenerateConfigurationProps.cs]: See https://github.com/dotnet/corefx/blob/master/src/Tools/GenerateProps/properties.props#L26

Naming is hard.  Wes and I spent some time trying to come up with this terminology.

**Comment by ericstj** [comment]
Merging this in to get a good package build.

File: src/Tools/CoreFx.Tools/GenerateConfigurationProps.cs (hunk 1 above)
Comment (joperezr) [inline src/Tools/CoreFx.Tools/GenerateConfigurationProps.cs]: I see that makes sense.

**Comment by joperezr** [comment]
LGTM

File: src/Tools/CoreFx.Tools/GenerateConfigurationProps.cs [hunk 2]
```diff
@@ -101,6 +101,20 @@ private void ParseProperties(string propertyName, ProjectRootElement project, bo
...
+                {
+                    // for insignificant properties, don't overwrite with default if already set, but derive values from it.
```
Comment (weshaggard) [inline src/Tools/CoreFx.Tools/GenerateConfigurationProps.cs]: While this approach works wouldn't it be easier to just add a not set condition on the setting of the default `'$(ArchGroup)'==''` ?

Comment (ericstj) [inline src/Tools/CoreFx.Tools/GenerateConfigurationProps.cs]: No because if there happened to be derived property values those wouldn't be set.

**Pull request merged**